tests/readPreference/readpreference-serialization-001.phpt

```diff
@@ -0,0 +1,187 @@
+--TEST--
+MongoDB\Driver\ReadPreference serialization
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+$tests = [
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_PRIMARY),
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_PRIMARY_PREFERRED),
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_SECONDARY),
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_SECONDARY_PREFERRED),
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_NEAREST),
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_PRIMARY, []),
```

Comment:
Would it make sense to also use `RP_SECONDARY` here? I realize an empty tag set is equivalent to omitting this argument entirely, so this is slightly pedantic. I also acknowledge that `mongoc_read_prefs_new()` and `mongoc_read_prefs_set_tags()` will both just call `bson_init` on the struct's `tags` field.